### PR TITLE
fix #12740

### LINKED ISSUE #12740: for loop tuple syntax not rendered correctly
### Example
```nim
runnableExamples:
  var data = @["one", "two"]
  for (i, d) in data.pairs:
    echo d
```

### Current Output
Syntax error
the temp file contains:
```nim
var data = @["one", "two"]
for (i) = in data.pairs:
  echo d
```

### Expected Output
nim doc compile and run correctly

```
$ nim -v
Nim Compiler Version 1.0.2
```

### PR DESCRIPTION
**Pull request by krux02**

### CODE REVIEW

File: tests/macros/tastrepr.nim [hunk 1]
```diff
@@ -0,0 +1,17 @@
+discard """
+output: '''
+var data = @["one", "two"]
+for (i, d) = in pairs(data):
```
Comment (cooldome) [inline tests/macros/tastrepr.nim]: equals sign `=` should not be here

Comment (cooldome) [reply tests/macros/tastrepr.nim]: Looks like gfor() needs to be update to handle tuple unpacking directly instead of relying on gsub()

Comment (krux02) [reply tests/macros/tastrepr.nim]: oh year, good catch. I was blind about that.

Comment (krux02) [reply tests/macros/tastrepr.nim]: @cooldome I thought so as well, but tuples can be recursive. I implement a different hack now.

**Comment by krux02** [comment]
This solution is a hack. It doesn't work for untyped ast. I tried to add a flag to the context, but the context is refreshed and as soon as the nkVarTuple is reached, the flags are gone again. Feedback is requested.

**Pull request merged**